packages/client/src/base/editor-context-service.ts

```diff
@@ -196,6 +205,21 @@ export class EditorContextService implements IActionHandler, Disposable, IDiagra
         this.onModelRootChangedEmitter.fire(root);
     }
 
+    /**
+     * Notifies the service about a viewport change. This method should not be called
+     * directly. It is called by the `CommandStack` after a viewport modifying command has been processed.
+     * @throws an error if the notifier is not a `CommandStack`
+     * @param viewport the new viewport
+     * @param notifier the object that triggered the viewport change
+     */
+    notifyViewportChanged(viewport: Readonly<Viewport>, notifier: AnyObject): void {
+        if (!(notifier instanceof CommandStack)) {
+            throw new Error('Invalid viewport change notification. Notifier is not an instance of `CommandStack`.');
```

Comment:
Out of curiosity, what is the reason for this restriction? Seems odd that adopters cannot trigger the notification but I see it is the same as in the model root change.